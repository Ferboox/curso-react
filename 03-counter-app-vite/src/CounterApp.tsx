import { useState } from 'react';

type Props = {
    value: number;
}

export const CounterApp = ({value}: Props) => {

    const [counter, setCounter] = useState( value );

    const handleAdd = (action:boolean) =>{
        const num = action ? 1 : -1;

        setCounter(counter + num)
    }

    const handleReset = () => setCounter(value)

    return(
        <>
            <h1>Counnter app</h1>
            <p>{ counter }</p>

            <button onClick={ () => handleAdd(true) }>+1</button>
            <button onClick={ () => handleAdd(false)}>-1</button>
            <button onClick={ () => handleReset() }>Reset</button>

        </>
    )
}
