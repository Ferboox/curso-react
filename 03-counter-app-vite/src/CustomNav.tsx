import { useEffect, useRef } from "react"

import { NavItem } from "./interfaces/interfaces";

type Props = {
    packageItems: NavItem[],
}

export const CustomNav = ({packageItems}:Props) => {

    
    const references = [
        useRef(null),
        useRef(null)
    ]

    const scrollToSection = (index:number) => {
        if(references[index].current != null){
            window.scrollTo({
                top: references[index].current.offsetTop,
                behavior: 'smooth'
            })
        }
    };

    console.log(references);

    return (
        <div id="customNav">
            <ul>
            {packageItems.map( (item, index) => (
                    <li key={item.name} onClick={() => scrollToSection(index)}>{item.name}</li>
                ))
            }
            </ul>
            <div ref={references[0]} style={{height: "80vh", background: "blue"}}>

            </div>
            <div ref={references[1]} className="contact" style={{height: "80vh", background: "yellow"}}>

            </div>
        </div>
    )
}

