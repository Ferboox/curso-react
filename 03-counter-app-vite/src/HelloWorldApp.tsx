export const App = () => {
    return(
        <h1>Hola mudedeedndo</h1>
    )
}