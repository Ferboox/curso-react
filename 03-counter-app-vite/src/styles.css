html, body{
    /* background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px; */
    margin: 0;
    box-sizing: border-box;
}

/* button{
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 5px;
}
 */

#customNav{
    width: 100%;
    background: red;
    padding: 0.5em 0.25rem;
}

#customNav ul{
    list-style: none;
    margin: 0;
    padding: 0;
    overflow: hidden;

}

#customNav ul li{
    float: left;
    margin-right: 1em;
    cursor: pointer;
}

#customNav ul li a {
    display: block;
    color: white;
    text-align: center;
    text-decoration: none;
}