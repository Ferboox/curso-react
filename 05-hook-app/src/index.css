body{
    padding: 20px;
}

button{
    margin-right: 1px;
}

a{
    margin-right: 5px;
}