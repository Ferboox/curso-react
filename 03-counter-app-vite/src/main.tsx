import React, { useRef } from "react";
import ReactDOM from "react-dom/client"
import {App} from "./HelloWorldApp";
import {FirstApp} from "./FirstApp";
import { CounterApp } from "./CounterApp";

import { CustomNav } from "./CustomNav";
import { NavItem } from "./interfaces/interfaces";

import './styles.css';

const items:NavItem[] = [
    {
        name: "Contacto",
        key: "contact"
    },
    {
        name: "Servicios",
        key: 'services'
    }
]



ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
    <React.StrictMode>
        {/* <App></App> */}
        {/* <FirstApp title="Hola, Soy Vegeta"/> */}
        <CounterApp value={ 10 }></CounterApp>
        {/* <CustomNav packageItems={items}></CustomNav> */}
       
    </React.StrictMode>
);