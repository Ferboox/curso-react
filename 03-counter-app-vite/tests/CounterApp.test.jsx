import { render, screen, fireEvent }from "@testing-library/react";
import { CounterApp } from '../src/CounterApp';

describe('Pruebas en <CounterApp/>', () => { 

  const value = 100;

  test('debe de hacer match con el snapshot', () => {

    const {container } = render (<CounterApp value={value}/>)
    expect( container ).toMatchSnapshot();

  })

  test('debe de mostrar el valor inicial "100"', () => {

    render( <CounterApp value={value} />)
    expect( screen.getByText(value) ).toBeTruthy()
  })

  test('debe de incrementar con el botón +1', () => {

    render( <CounterApp value={value} />)
    fireEvent.click( screen.getByText('+1') )

    expect( screen.getByText('101')).toBeTruthy();
  })

  test('debe de decrementar con el botón -1', () => {

    render( <CounterApp value={value} />)
    fireEvent.click( screen.getByText('-1') )

    expect( screen.getByText('99')).toBeTruthy();
  })


  test('debe de reiniciar con el botón "Reset"', () => {

    render( <CounterApp value={value} />)

    fireEvent.click( screen.getByText('-1') )
    fireEvent.click( screen.getByText('Reset') )

    expect( screen.getByText(value)).toBeTruthy();
  })


});