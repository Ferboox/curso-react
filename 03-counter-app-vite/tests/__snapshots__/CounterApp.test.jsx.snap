// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counnter app
  </h1>
  <p>
    100
  </p>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
